ocaml/testsuite/tests/typing-modes/modes.ml

```diff
@@ -57,3 +58,89 @@ Error: Unrecognized mode foo for coercion.
 (* CR zqian: this should be supported *)
 (* let foo a b @ local = "hello"
 let foo a b : _ @@ local = "hello" *)
+
+(* Expressions *)
+let foo = ("hello" : _ @@ local)
+[%%expect{|
+Line 1, characters 10-32:
+1 | let foo = ("hello" : _ @@ local)
+              ^^^^^^^^^^^^^^^^^^^^^^
+Error: This value escapes its region
+|}]
+
+(* this is not mode annotation *)
+let foo = ("hello" @ local)
+[%%expect{|
+Line 1, characters 11-18:
+1 | let foo = ("hello" @ local)
+               ^^^^^^^
+Error: This expression has type string but an expression was expected of type
+         'a list
+|}]
+
+let foo ~bar = bar ^ "hello"
+
+let x =
+  let bar = "world" in
+  foo ~(bar : _ @@ unique)
+[%%expect{|
+val foo : bar:string -> string = <fun>
+Line 5, characters 8-11:
+5 |   foo ~(bar : _ @@ unique)
+            ^^^
+Error: Found a shared value where a unique value was expected
+|}]
+
+let x =
+  let bar = "world" in
+  foo ~(bar @ unique)
+[%%expect{|
+Line 3, characters 8-11:
+3 |   foo ~(bar @ unique)
+            ^^^
+Error: Found a shared value where a unique value was expected
+|}]
+
+type r = {a : string; b : string}
+
+let r = {a : _ @@ unique = "hello";
+         b : _ @@ once = "world"}
```

Comment:
Indeed this is not in the design document. The reason I did this is similar to the one below: because it's more consistent and it's simple.

In term of confusion with modalities - modalities can only show up in type declarations - do we think the users will know that?